plugins/modules/platform_update.py

```diff
@@ -16,21 +16,27 @@
 module: platform_update
 author:
     - Chiranthan M V (@chiranthanmv)
-short_description: Performs consolidated system firmware, VIOS, SR-IOV, and I/O adapter updates, including partition migration.
+short_description: Performs Consolidated system firmware (update/upgrade), VIOS, SR-IOV, and I/O adapter updates, including optional partition migration.
 notes:
-  - The current version supports IBM Fix Central website as the update source .
-  - Support for additional update sources will be added in future releases.
-  - For any update operation, if the Ansible response status is C(ok) with C(changed) set to C(false)
-    and the result is C(COMPLETED_WITH_ERROR) with the reason C("update not available"),
-    it means the target is already up-to-date.
-description:
-    - This module allows updating System Firmware, VIOS, SR-IOV adapters, and I/O adapters either individually or through a single consolidated update flow.
-    - This operations can be configured using I(platform_config) parameter.
-    - It supports both 'DriverOnly' and 'Adapter' update strategies for SR-IOV adapters and allows updates from IBM Fix Central website when specified.
-    - Supports defining the order in which updates are applied across components.
+  - The current version supports IBM Fix Central website as the update/upgrade source .
+  - Support for additional update/upgrade sources will be added in future releases.
+  - Supports defining the order in which update/upgrade are applied across components.
+  - To perform configuration operations, you do not need to specify a separate state or action.
+    Supplying values under C(platform_config) is sufficient to apply the changes directly to the HMC.
+  - When performing an update or upgrade operation via C(IBMWebsite) with C(level='latest'),
+    if the Ansible response status is C(ok) and C(changed) is C(false), and the result is C(COMPLETED_WITH_ERROR) with the reason C("update not available"),
+    it indicates that no newer update images are available or the target is already up-to-date.
+  - The module supports idempotency for System Firmware Update/Upgrade, VIOS Update, SR-IOV Adapter Update, and I/O Adapter Update operations.
```

Comment:
Are you sure idempotency is supported for all? are you comparing the current version and user supplied version before supplying the details to rest call?